cpp/tests/copying/scatter_tests.cu

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <thrust/device_vector.h>
+#include "copying.hpp"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "tests/utilities/column_wrapper.cuh"
+#include "tests/utilities/cudf_test_fixtures.h"
+#include "tests/utilities/cudf_test_utils.cuh"
+#include "types.hpp"
+
+template <typename T>
+struct ScatterTest : GdfTest {};
+
+using test_types = ::testing::Types<int8_t, int16_t, int32_t, int64_t, float, double>;
+TYPED_TEST_CASE(ScatterTest, test_types);
+
```

Comment:
Shouldn't you test each of the error cases for correct error return values? (e.g. mismatched sizes, data types, etc?)